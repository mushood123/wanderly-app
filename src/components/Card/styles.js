import {StyleSheet} from 'react-native';
import {font} from '../../theme/fonts';

export const styles = StyleSheet.create({
  card: {
    backgroundColor: 'powderblue',
    borderRadius: 10,
    padding: 15,
    margin: 10,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
  header: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 10,
  },
  avatar: {
    width: 60,
    height: 60,
    borderRadius: 30,
    marginRight: 15,
  },
  ranking: {
    backgroundColor: '#f0f0f0',
    padding: 5,
    borderRadius: 5,
    marginRight: 'auto',
  },
  rankingText: {
    fontWeight: 'bold',
  },
  content: {
    marginBottom: 10,
  },
  name: {
    fontFamily: font.heading,
    fontSize: 18,
    fontWeight: 'bold',
    marginBottom: 5,
  },
  title: {
    fontSize: 15,
    marginBottom: 2,
  },
  stats: {
    flexDirection: 'column',
    marginLeft: 10,
    justifyContent: 'space-between',
  },
  stat: {
    textAlign: 'center',
  },
  statValue: {
    fontFamily: font.heading,
    fontSize: 16,
    fontWeight: 'bold',
    marginBottom: 2,
  },
  statLabel: {
    fontSize: 15,
  },
  saperator: {
    height: '100%',
    width: 1,
    backgroundColor: '#918d8d',
    marginHorizontal: 10,
  },
});
