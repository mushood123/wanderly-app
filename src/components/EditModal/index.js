import { EditModal } from './editModal';

export { EditModal };
