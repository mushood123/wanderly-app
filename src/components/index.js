import {FormField} from './FormField';
import {Text} from './Text';

export {FormField, Text};
