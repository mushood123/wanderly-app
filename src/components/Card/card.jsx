import React, {useContext} from 'react';
import {TouchableOpacity, View} from 'react-native';
import {Text} from '../Text';
import {styles} from './styles';
import {LanguageContext} from '../../contexts';
import {AuthContext} from '../../contexts';
import {
  CardContainer,
  CardContent,
  UserContent,
  UserName,
  Separator,
} from './styles';

export const Card = ({
  hourlyRate = 0,
  places = [],
  name = 'NA',
  place = 'Lahore',
  currentUserId = 0,
  uid,
  showButton,
  onEditPressed = () => {},
  onBuyPressed = () => {},
  packageId,
  offer,
}) => {
  const {user} = useContext(AuthContext);
  const {locale} = useContext(LanguageContext);
  return (
    <CardContainer>
      <CardContent>
        <UserContent>
          <UserName>{name}</UserName>
          <Text>{place}</Text>
        </UserContent>
        <Separator />

        <View style={styles.stats}>
          <Text style={styles.statValue}>{locale.CLAUSE.VISITING_PLACES}</Text>

          {places.map((place, index) => {
            return (
              <Text key={index} style={styles.title}>
                {'-> '}
                {place}
              </Text>
            );
          })}

          <Text style={styles.stat}>
            <Text style={styles.statValue}> {locale.CLAUSE.HOURLY_RATE}: </Text>
            <Text style={styles.statLabel}>{hourlyRate}</Text>
          </Text>
        </View>
      </CardContent>
      {showButton && (
        <TouchableOpacity
          onPress={() => {
            if (uid === currentUserId) {
              onEditPressed({...offer, packageId});
            } else {
              onBuyPressed(packageId, user.uid);
            }
          }}
          style={styles.btn}>
          <Text style={styles.statValue}>
            {uid === currentUserId ? locale.LABEL.EDIT : locale.LABEL.BUY}
          </Text>
        </TouchableOpacity>
      )}
    </CardContainer>
  );
};
