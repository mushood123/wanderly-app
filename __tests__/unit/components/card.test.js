import React from 'react';
import { render } from '../utils/theme-wrapper/theme-wrapper';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { Card } from '~src/components/Card';


const mockStore = configureStore([]);

describe('Card component', () => {
    let store;

    beforeEach(() => {
        store = mockStore({
            auth: {
                user: {
                    uid: 'user123',
                },
            },
            language: {
                locale: {
                    CLAUSE: {
                        VISITING_PLACES: 'Visiting Places',
                        HOURLY_RATE: 'Hourly Rate',
                    },
                    LABEL: {
                        EDIT: 'Edit',
                        BUY: 'Buy',
                    },
                },
            },
        });
    });

    it('renders correctly and matches snapshot with default props', () => {
        const tree = render(
            <Provider store={store}>
                <Card />
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with custom props and matches snapshot', () => {
        const customProps = {
            hourlyRate: 50,
            places: ['New York', 'Los Angeles'],
            name: 'John Doe',
            place: 'San Francisco',
            currentUserId: 'user123',
            uid: 'user456',
            showButton: true,
            onEditPressed: jest.fn(),
            onBuyPressed: jest.fn(),
            packageId: 'package123',
            offer: { title: 'Custom Offer' },
        };

        const tree = render(
            <Provider store={store}>
                <Card {...customProps} />
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with edit button when user owns the package', () => {
        const customProps = {
            currentUserId: 'user123', // Same as logged-in user
            uid: 'user123', // Same as currentUserId to simulate package owner
            showButton: true,
            packageId: 'package123',
            offer: { title: 'Custom Offer' },
        };

        const tree = render(
            <Provider store={store}>
                <Card {...customProps} />
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with buy button for another user', () => {
        const customProps = {
            currentUserId: 'user123',
            uid: 'user789', // Different user
            showButton: true,
            packageId: 'package123',
            offer: { title: 'Custom Offer' },
        };

        const tree = render(
            <Provider store={store}>
                <Card {...customProps} />
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
