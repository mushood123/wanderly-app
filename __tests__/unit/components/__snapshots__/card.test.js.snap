// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders correctly and matches snapshot with default props 1`] = `
<View
  shadowOffset={
    {
      "height": 2,
      "width": 0,
    }
  }
  style={
    {
      "backgroundColor": "#A1E3F7",
      "borderBottomLeftRadius": 10,
      "borderBottomRightRadius": 10,
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
      "elevation": 5,
      "marginBottom": 10,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 10,
      "paddingBottom": 15,
      "paddingLeft": 15,
      "paddingRight": 15,
      "paddingTop": 15,
      "shadowColor": "#000",
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 10,
      }
    }
  >
    <View
      style={
        {
          "marginBottom": 10,
        }
      }
    >
      <Text
        style={
          {
            "fontFamily": "Chella-Bold",
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 5,
          }
        }
      >
        NA
      </Text>
      <Text
        style={
          {
            "fontFamily": "Poppins-Regular",
          }
        }
      >
        Lahore
      </Text>
    </View>
    <View
      style={
        {
          "backgroundColor": "#918d8d",
          "height": "100%",
          "marginHorizontal": 10,
          "width": 1,
        }
      }
    />
    <View
      style={
        {
          "flexDirection": "column",
          "justifyContent": "space-between",
          "marginLeft": 10,
        }
      }
    >
      <Text
        style={
          {
            "fontFamily": "Chella-Bold",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginBottom": 2,
          }
        }
      >
        Visiting Places
      </Text>
      <Text
        style={
          {
            "textAlign": "center",
          }
        }
      >
        <Text
          style={
            {
              "fontFamily": "Chella-Bold",
              "fontSize": 16,
              "fontWeight": "bold",
              "marginBottom": 2,
            }
          }
        >
           
          Hourly Rate
          : 
        </Text>
        <Text
          style={
            {
              "fontSize": 15,
            }
          }
        >
          0
        </Text>
      </Text>
    </View>
  </View>
</View>
`;

exports[`Card component renders correctly with buy button for another user 1`] = `
<View
  shadowOffset={
    {
      "height": 2,
      "width": 0,
    }
  }
  style={
    {
      "backgroundColor": "#A1E3F7",
      "borderBottomLeftRadius": 10,
      "borderBottomRightRadius": 10,
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
      "elevation": 5,
      "marginBottom": 10,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 10,
      "paddingBottom": 15,
      "paddingLeft": 15,
      "paddingRight": 15,
      "paddingTop": 15,
      "shadowColor": "#000",
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 10,
      }
    }
  >
    <View
      style={
        {
          "marginBottom": 10,
        }
      }
    >
      <Text
        style={
          {
            "fontFamily": "Chella-Bold",
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 5,
          }
        }
      >
        NA
      </Text>
      <Text
        style={
          {
            "fontFamily": "Poppins-Regular",
          }
        }
      >
        Lahore
      </Text>
    </View>
    <View
      style={
        {
          "backgroundColor": "#918d8d",
          "height": "100%",
          "marginHorizontal": 10,
          "width": 1,
        }
      }
    />
    <View
      style={
        {
          "flexDirection": "column",
          "justifyContent": "space-between",
          "marginLeft": 10,
        }
      }
    >
      <Text
        style={
          {
            "fontFamily": "Chella-Bold",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginBottom": 2,
          }
        }
      >
        Visiting Places
      </Text>
      <Text
        style={
          {
            "textAlign": "center",
          }
        }
      >
        <Text
          style={
            {
              "fontFamily": "Chella-Bold",
              "fontSize": 16,
              "fontWeight": "bold",
              "marginBottom": 2,
            }
          }
        >
           
          Hourly Rate
          : 
        </Text>
        <Text
          style={
            {
              "fontSize": 15,
            }
          }
        >
          0
        </Text>
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderBottomLeftRadius": 10,
        "borderBottomRightRadius": 10,
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "justifyContent": "center",
        "marginVertical": 10,
        "opacity": 1,
        "paddingVertical": 20,
      }
    }
  >
    <Text
      style={
        {
          "fontFamily": "Chella-Bold",
          "fontSize": 16,
          "fontWeight": "bold",
          "marginBottom": 2,
        }
      }
    >
      Buy
    </Text>
  </View>
</View>
`;

exports[`Card component renders correctly with custom props and matches snapshot 1`] = `
<View
  shadowOffset={
    {
      "height": 2,
      "width": 0,
    }
  }
  style={
    {
      "backgroundColor": "#A1E3F7",
      "borderBottomLeftRadius": 10,
      "borderBottomRightRadius": 10,
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
      "elevation": 5,
      "marginBottom": 10,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 10,
      "paddingBottom": 15,
      "paddingLeft": 15,
      "paddingRight": 15,
      "paddingTop": 15,
      "shadowColor": "#000",
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 10,
      }
    }
  >
    <View
      style={
        {
          "marginBottom": 10,
        }
      }
    >
      <Text
        style={
          {
            "fontFamily": "Chella-Bold",
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 5,
          }
        }
      >
        John Doe
      </Text>
      <Text
        style={
          {
            "fontFamily": "Poppins-Regular",
          }
        }
      >
        San Francisco
      </Text>
    </View>
    <View
      style={
        {
          "backgroundColor": "#918d8d",
          "height": "100%",
          "marginHorizontal": 10,
          "width": 1,
        }
      }
    />
    <View
      style={
        {
          "flexDirection": "column",
          "justifyContent": "space-between",
          "marginLeft": 10,
        }
      }
    >
      <Text
        style={
          {
            "fontFamily": "Chella-Bold",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginBottom": 2,
          }
        }
      >
        Visiting Places
      </Text>
      <Text
        style={
          {
            "fontSize": 15,
            "marginBottom": 5,
          }
        }
      >
        -&gt; 
        New York
      </Text>
      <Text
        style={
          {
            "fontSize": 15,
            "marginBottom": 5,
          }
        }
      >
        -&gt; 
        Los Angeles
      </Text>
      <Text
        style={
          {
            "textAlign": "center",
          }
        }
      >
        <Text
          style={
            {
              "fontFamily": "Chella-Bold",
              "fontSize": 16,
              "fontWeight": "bold",
              "marginBottom": 2,
            }
          }
        >
           
          Hourly Rate
          : 
        </Text>
        <Text
          style={
            {
              "fontSize": 15,
            }
          }
        >
          50
        </Text>
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderBottomLeftRadius": 10,
        "borderBottomRightRadius": 10,
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "justifyContent": "center",
        "marginVertical": 10,
        "opacity": 1,
        "paddingVertical": 20,
      }
    }
  >
    <Text
      style={
        {
          "fontFamily": "Chella-Bold",
          "fontSize": 16,
          "fontWeight": "bold",
          "marginBottom": 2,
        }
      }
    >
      Buy
    </Text>
  </View>
</View>
`;

exports[`Card component renders correctly with edit button when user owns the package 1`] = `
<View
  shadowOffset={
    {
      "height": 2,
      "width": 0,
    }
  }
  style={
    {
      "backgroundColor": "#A1E3F7",
      "borderBottomLeftRadius": 10,
      "borderBottomRightRadius": 10,
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
      "elevation": 5,
      "marginBottom": 10,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 10,
      "paddingBottom": 15,
      "paddingLeft": 15,
      "paddingRight": 15,
      "paddingTop": 15,
      "shadowColor": "#000",
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 10,
      }
    }
  >
    <View
      style={
        {
          "marginBottom": 10,
        }
      }
    >
      <Text
        style={
          {
            "fontFamily": "Chella-Bold",
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 5,
          }
        }
      >
        NA
      </Text>
      <Text
        style={
          {
            "fontFamily": "Poppins-Regular",
          }
        }
      >
        Lahore
      </Text>
    </View>
    <View
      style={
        {
          "backgroundColor": "#918d8d",
          "height": "100%",
          "marginHorizontal": 10,
          "width": 1,
        }
      }
    />
    <View
      style={
        {
          "flexDirection": "column",
          "justifyContent": "space-between",
          "marginLeft": 10,
        }
      }
    >
      <Text
        style={
          {
            "fontFamily": "Chella-Bold",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginBottom": 2,
          }
        }
      >
        Visiting Places
      </Text>
      <Text
        style={
          {
            "textAlign": "center",
          }
        }
      >
        <Text
          style={
            {
              "fontFamily": "Chella-Bold",
              "fontSize": 16,
              "fontWeight": "bold",
              "marginBottom": 2,
            }
          }
        >
           
          Hourly Rate
          : 
        </Text>
        <Text
          style={
            {
              "fontSize": 15,
            }
          }
        >
          0
        </Text>
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderBottomLeftRadius": 10,
        "borderBottomRightRadius": 10,
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "justifyContent": "center",
        "marginVertical": 10,
        "opacity": 1,
        "paddingVertical": 20,
      }
    }
  >
    <Text
      style={
        {
          "fontFamily": "Chella-Bold",
          "fontSize": 16,
          "fontWeight": "bold",
          "marginBottom": 2,
        }
      }
    >
      Edit
    </Text>
  </View>
</View>
`;
